src/kartograf/atom_mapper.py

```diff
@@ -855,6 +854,38 @@ def suggest_mapping_from_rdmols(
 
         return mapping
 
+    def _split_protein_components_molecules(self, protein: ProteinComponent) -> list[ProteinComponent]:
```

Comment:
If we're not using `self`, maybe make this a staticmethod so it can be tested a bit more easily?